### PR TITLE
P chain import, add validator, and add delegator transactions.

### PR DESCRIPTION
**Pull request by jonored**

### CODE REVIEW

**Comment by jonored** [comment]
This still has an X-macro network ID lookup that needs removed, and we still need some tests on _x-chain_ import and export as there weren't any in the suite and I changed things.

**Comment by jonored** [comment]
Macro removed, import/export on X-chain tested, and tests updated for merge from develop.

File: src/parser.c [hunk 1]
```diff
@@ -420,6 +531,7 @@ static void strcpy_prompt(char *const out, size_t const out_size, char const *co
 
 static bool prompt_fee(parser_meta_state_t *const meta) {
     uint64_t fee = -1; // if this is unset this should be obviously wrong
+    PRINTF("inputs: %.*h outpfts: %.*h\n", 8, &meta->sum_of_inputs, 8, &meta->sum_of_outputs);
```
Comment (matthewbauer) [inline src/parser.c:534]:
```suggestion
    PRINTF("inputs: %.*h outputs: %.*h\n", 8, &meta->sum_of_inputs, 8, &meta->sum_of_outputs);
```

File: src/parser.c [hunk 2]
```diff
@@ -430,114 +542,103 @@ static bool prompt_fee(parser_meta_state_t *const meta) {
...
+      return false;
+  }
+}
+
+enum parse_rv parse_BaseTransaction(struct BaseTransactionState *const state, parser_meta_state_t *const meta) {
     enum parse_rv sub_rv = PARSE_RV_INVALID;
     switch (state->state) {
-        case 2: { // Network ID
-            INIT_SUBPARSER(uint32State, uint32_t);
+        case 0: { // Network ID
```
Comment (matthewbauer) [inline src/parser.c:565]: Since these are changing, can we make an enum / #define for it?

File: tests/basic-tests.js [hunk 3]
```diff
@@ -466,20 +462,164 @@ describe("Basic Tests", () => {
       );
     });
   });
-});
+  describe("X-chain Import and Export", function() {
+    it("Can sign an Import transaction", async function() {
+    const txn = Buffer.from([
```
Comment (matthewbauer) [inline tests/basic-tests.js:467]: indentation is off here

File: src/parser.c [hunk 4]
```diff
@@ -647,23 +817,70 @@ enum parse_rv parseTransaction(struct TransactionState *const state, parser_meta
...
+              case TRANSACTION_TYPE_ID_PLATFORM_EXPORT:
+                CALL_SUBPARSER(exportTxState, ExportTransaction);
                 break;
-
               default:
                 REJECT("Only base, export, and import transactions are supported");
             }
+            BUBBLE_SWITCH_BREAK;
+            state->state++;
+                }
```
Comment (matthewbauer) [inline src/parser.c:875]: indentation off here too

**Review by matthewbauer — APPROVED** [review]

**Pull request merged**